youtube_dl/extractor/olympicchannel.py

```diff
@@ -0,0 +1,50 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import parse_duration, unified_strdate
+
+
+class OlympicChannelIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?olympicchannel\.com/../playback/(?P<id>.*)'
+    _TEST = {
+        'url': 'https://www.olympicchannel.com/en/playback/snowboard-mens-halfpipe',
+        'info_dict': {
+            'id': 'epis_63b711050c9749ab8ca3a6fa07f4cff6',
+            'ext': 'mp4',
+            'title': 'Men\'s Halfpipe Finals - Snowboard | PyeongChang 2018 Replays',
+            'description': 'The men\'s halfpipe finals were held at the Phoenix Snow Park on 14 February 2018.',
+            'duration': 6780,
+            'release_date': '20160601',
+            'thumbnail': 'https://images.olympicchannel.com/media/images/stills/content/56721/a929eb66b550de98a453ef4d87c7856c.jpg'
+        }
+    }
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        data = self._parse_json(
+            self._html_search_regex(r'data-widget-data=["\']([^"\']*)["\']', webpage, 'data'), display_id)
+        video_id = data['uid']
+        title = data.get('title')
+        description = data.get('synopsis')
+        thumbnail = data.get('images', {}).get('main', {}).get('url')
+        formats = None
```

Comment:
Must not be `None`.

Reply:
I figured it would be caught as an error later on, but an explicit error message is probably better so I added one.